src/btree/bt_sync.c

```diff
@@ -122,27 +137,41 @@ __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 static int
 __sync_ref_mark_deleted(WT_SESSION_IMPL *session, WT_REF *ref)
 {
-    /* Ignore root pages as they are never be deleted */
-    if (__wt_ref_is_root(ref))
-        return (0);
-
-    /* Guard against marking a page to be deleted when its reconciliation state is not empty. */
-    if (ref->page != NULL && ref->page->modify != NULL &&
-      ref->page->modify->rec_result != WT_PM_REC_EMPTY)
-        return (0);
-
     /*
      * Mark the page as deleted and also set the parent page as dirty. This is to ensure when the
      * parent page is checkpointing, the empty child page will be cleaned.
      */
-    if (WT_REF_CAS_STATE(session, ref, ref->state, WT_REF_LOCKED)) {
+    if (WT_REF_CAS_STATE(session, ref, WT_REF_DISK, WT_REF_LOCKED)) {
```

Comment:
I think the sentence `This is to ensure when the parent page is checkpointing, the empty child page will be cleaned.` is confusing, what do you think of `This is to ensure the parent page must be written during checkpoint and the child page discarded.`

Reply:
Used the provided sentence.